internal/pkg/describe/uri.go

```diff
@@ -13,13 +13,21 @@ import (
 	"github.com/aws/copilot-cli/internal/pkg/manifest"
 )
 
+type URIAccessType int
+
+const (
+	URIAccessTypeInternet URIAccessType = iota + 1
+	URIAccessTypeInternal
+	URIAccessTypeServiceDiscovery
+)
+
 var (
 	fmtSvcDiscoveryEndpointWithPort = "%s.%s:%s" // Format string of the form {svc}.{endpoint}:{port}
 )
 
-// ReachableService represents a service describe that has an endpoint.
+// ReachableService represents a service describer that has an endpoint.
 type ReachableService interface {
-	URI(env string) (string, error)
+	URI(env string) (string, URIAccessType, error)
```

Comment:
Can we return a wrapped-up type so that we don't need to return three values for this function?

Reply:
🥺

Reply:
I'm also a little worried with the 3 return types

I haven't read the PR so take my feedback with a grain of salt 😂 but I wonder if we can have something like this: https://go.dev/play/p/UeAeOWHH7nT 💭

Reply:
lmk if you have a better idea for the struct name :)